samcli/hook_packages/terraform/hooks/prepare/resource_linking.py

```diff
@@ -69,6 +69,115 @@ def full_address(self) -> str:
         return self.address
 
 
+def _build_module(
+    module_name: Optional[str],
+    module_configuration: Dict,
+    input_variables: Dict[str, Expression],
+    parent_module_address: Optional[str],
+) -> TFModule:
+    module = TFModule(None, None, {}, [], {}, {})
+
+    module.full_address = _build_module_full_address(module_name, parent_module_address)
+    module.variables = _build_module_variables_from_configuration(module_configuration, input_variables)
+    module.resources = _build_module_resources_from_configuration(module_configuration, module)
+    module.outputs = _build_module_outputs_from_configuration(module_configuration)
+    module.child_modules = _build_child_modules_from_configuration(module_configuration, module)
+
+    return module
+
+
+def _build_module_full_address(module_name: Optional[str], parent_module_address: Optional[str]) -> Optional[str]:
+    full_address = None
```

Comment:
so the root module address will be None not an empty string?

Reply:
yep